### PR TITLE
Kafka improvements

### PR DESCRIPTION
**Pull request by nathan-artie**
- Add a function to build a Kafka writer from `config.Kafka`
- Add a reloadable Kafka writer
- Add `NewMessage` function
- Refactor dynamo.KafkaMessage to use `NewMessage`

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -19,6 +20,10 @@ type Kafka struct {
 	MaxRequestSize   uint64 `yaml:"maxRequestSize,omitempty"`
 }
 
+func (k *Kafka) BootstrapAddresses() []string {
```
Comment (nathan-artie) [inline config/config.go:23]: From https://github.com/artie-labs/reader/pull/38/files#diff-fa1424093931eae4e4acefbba233db02bd422e6acf68476c00295ce96cd11b34L43

File: lib/dynamo/message.go [hunk 2]
```diff
@@ -109,22 +109,5 @@ func (m *Message) KafkaMessage(kafkaCfg config.Kafka) (kafka.Message, error) {
 	if err != nil {
 		return kafka.Message{}, fmt.Errorf("failed to generate artie message, err: %v", err)
 	}
-
-	jsonBytes, err := json.Marshal(msg)
```
Comment (nathan-artie) [inline lib/dynamo/message.go:113]: Moved to https://github.com/artie-labs/reader/pull/38/files#diff-c35d4608207032c28eeadafe90e54346786b4b8dee23ea55e1c77118bd9ac2feR10

File: lib/kafkalib/message.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package kafkalib
+
+import (
+	"encoding/json"
+
+	"github.com/artie-labs/transfer/lib/cdc/util"
+	"github.com/segmentio/kafka-go"
+)
+
+func NewMessage(topic string, key map[string]interface{}, value util.SchemaEventPayload) (kafka.Message, error) {
```
Comment (Tang8330) [inline lib/kafkalib/message.go:10]: Call it `partitionKey`?

**Review by Tang8330 — APPROVED** [review]
lgtm

**Pull request merged**